qjs.c

```diff
@@ -371,12 +371,13 @@ int main(int argc, char **argv)
                 continue;
             }
             if (opt == 'd' || !strcmp(longopt, "dump")) {
+                dump_memory++;
                 if (opt_arg) {
                     dump_flags = strtol(opt_arg, NULL, 16);
                     break;
+                } else {
+                    continue;
                 }
-                dump_memory++;
-                continue;
             }
```

Comment:
There is a side effect here: `dump_memory` is always set, even if an argument is provided (is: `-d1`), which is not my intent.  I suggest a simpler fix:
```
-                 if (opt_arg) {
+                 if (opt_arg && isdigit((unsigned char)*opt_arg)) {
```

Reply:
I like the latter option, let's go for that!